### PR TITLE
 Made Json Datastore use Resource Selector.

### LINKED ISSUE #856: Make JSON datastore use ResourceSelector (like CSV datastore)

### PR DESCRIPTION
**Pull request by ClaudiaPHI**
Fixes #856.

### CODE REVIEW

File: desktop/ui/src/main/java/org/datacleaner/windows/JsonDatastoreDialog.java [hunk 1]
```diff
@@ -58,20 +58,30 @@ public String getWindowTitle() {
...
     }
+
+    @Override
+    protected boolean isPreviewTableEnabled() {
+        return true;
+    }
+
+    @Override
+    protected boolean isPreviewDataAvailable() {
+        return showPreview;
```
Comment (kaspersorensen) [inline desktop/ui/src/main/java/org/datacleaner/windows/JsonDatastoreDialog.java]: Does this ever change? I guess not. But I cannot remember what the purpose was from the super-class's side. Maybe we need to check something?

**Comment by kaspersorensen** [comment]
LGTM, just a question from my side...

File: desktop/ui/src/main/java/org/datacleaner/windows/JsonDatastoreDialog.java (hunk 1 above)
Comment (ClaudiaPHI) [inline desktop/ui/src/main/java/org/datacleaner/windows/JsonDatastoreDialog.java]: I checked and it seems not to be relevant to have a volatile. I removed it.

**Comment by ClaudiaPHI** [comment]
screen shot 2015-10-28 at 10 32 57 am

**Comment by LosD** [comment]
LGTM! Merge when Travis is happy

**Comment by LosD** [comment]
Travis loves it, merging :)

**Pull request merged**